src/reg_allocator.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+This is a part of Loops project.
+Distributed under Apache 2 license.
+See https://github.com/vpisarev/loops/LICENSE
+*/
+
+#ifndef __LOOPS_REG_ALLOCATOR_HPP__
+#define __LOOPS_REG_ALLOCATOR_HPP__
+
+#include "loops/loops.hpp"
+#include "backend.hpp"
+#include "common.hpp"
+#include <map>
+
+namespace loops {
+
+    class RegisterPool
+    {
+    public:
+        RegisterPool(ContextImpl* m_owner);
```

Comment:
I believe, at some point we will need to define internal API with a clear architecture description. It's not good that register allocator communicates with ContextImpl

Reply:
Is Backend* good enough?